Cython/Compiler/ExprNodes.py

```diff
@@ -13878,6 +13892,52 @@ def analyse_type_annotation(self, env, assigned_value=None):
             warning(annotation.pos, "Unknown type declaration in annotation, ignoring")
         return base_type, arg_type
 
+class AssignmentExpressionNode(ExprNode):
+    """
+    Also known as a named expression or the walrus operator
+
+    Arguments
+    lhs - NameNode
+    rhs - ExprNode  (not stored as an attribute because it is likely to be heavily transformed)
```

Comment:
If the RHS gets transformed, then the reference to it should change here. Why is that not enough?

Reply:
`rhs` ends up as a child of the `TempResultFromStatNode` and that's ultimately where it gets transformed. It's fiddly to keep in in sync with `rhs` as an attribute of `AssignmentExpressionNode` (and it isn't *that* useful here). If we want it accessible as an attribute it probably needs to be a property